mlir/test/tblgen-to-irdl/TestDialect.td

```diff
@@ -16,8 +16,11 @@ class Test_Type<string name, string typeMnemonic, list<Trait> traits = []>
 class Test_Op<string mnemonic, list<Trait> traits = []>
     : Op<Test_Dialect, mnemonic, traits>;
 
+// CHECK: irdl.type @"!singleton_a"
```

Comment:
The name should just be `@singleton_a` here.
The reason we had a `!` in some places was only in string names (compared to references), to distinguish types and attributes.

Reply:
You can still get name clashes in the symbol names. There are dialects which have operations, types, and attributes share name (I think stablehlo is an example)

Reply:
Oh wait you're right yes!